sources/platform/actors/development/builds_and_runs/state_persistence.md

```diff
@@ -1,51 +1,64 @@
 ---
 title: State persistence
-description: Maintain a long-running Actor's state to prevent unexpected restarts. See a code example on how to prevent a run in the case of a server shutdown.
+description: Learn how to maintain an Actor's state to prevent data loss during unexpected restarts. Includes code examples for handling server migrations.
 slug: /actors/development/builds-and-runs/state-persistence
 ---
 
 # State persistence
 
-**Maintain a long-running Actor's state to prevent unexpected restarts. See a code example on how to prevent a run in the case of a server shutdown.**
+**Learn how to maintain an Actor's state to prevent data loss during unexpected restarts. Includes code examples for handling server migrations.**
 
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
 ---
 
-Long-running [Actor](../../index.mdx) jobs may need to migrate from one server to another. Unless you save your job's progress, it will be lost during the migration. The Actor will restart from scratch on the new server, which can be costly.
+Long-running [Actor](../../index.mdx) jobs may need to migrate between servers. Without state persistence, your job's progress, is lost during migration, causing it to restart from the beginning on the new server. This can be costly and time-consuming.
 
-To avoid this, long-running Actors should save (persist) their state periodically and listen for [migration events](/sdk/js/api/apify/class/PlatformEventManager). When started, these Actors should [check for persisted state](#code-examples), so they can continue where they left off.
+To prevent data loss, long-running Actors should:
 
-For short-running Actors, the chance of a restart and the cost of repeated runs are low, so restarts can be ignored.
+- Periodically save (persist) their state.
+- Listem for [migration events](/sdk/js/api/apify/class/PlatformEventManager)
+- Check for persisted state when starting, allowing them to resume from where they left off.
 
-## What is a migration?
+For short-running Actors, the risk of restarts and the cost of repeated runs are low, so you can typically ignore state persistence.
 
-A migration is when a process running on a server has to stop and move to another. All in-progress processes on the current server are stopped. Unless you have saved your state, the Actor run will restart on the new server. For example, if a request in your [request queue](../../../storage/request_queue.md) has not been updated as **crawled** before the migration, it will be crawled again.
+## Undersanding migrations
 
-**When a migration event occurs, you only have a few seconds to save your work.**
+A migration occurs when a process running on one srever must stop and move to another. During this process:
 
-## Why do migrations happen
+- All in-progress processes on the current server are stopped
+- Unless you've saved your state, the Actor run will restart on the new server
```

Comment:
The Actor will restart even if you've saved your state, the only difference is how
```suggestion
- Unless you've saved your state, the Actor run will restart on the new server with an empty internal state
```